report_thesis/src/sections/background.tex

```diff
@@ -316,7 +316,59 @@ \subsubsection{Extra Trees Regressor (ETR)}
 As a tradeoff, \gls{etr} is less interpretable than a single decision tree, as the added randomness can introduce more bias than \gls{rf}.
 However, it often achieves better generalization performance, especially in high-dimensional or noisy datasets.
 
-\subsubsection{Gradient Boosting Regressor (GBR)}
+\subsubsection{Gradient Boosting Regression (GBR)}\label{sec:gradientboost}
+In this section we introduce \gls{gbr} primarily based on \citet{James2023AnIS}.
+\gls{gbr} is an ensemble learning technique that uses a sequence of weak learners, typically decision trees, to build a strong predictive model.
+
+However, in order to explain \gls{gbr}, it is helpful to build on the concepts of ensemble learning and decision trees.
+Ensemble learning is a technique in machine learning where multiple models, known as \textit{weak learners}, are combined to produce more accurate predictions.
+Mathematically, ensemble learning can be defined as combining the predictions of $M$ weak learners to form a final prediction $\hat{y}$, such that:
+\begin{equation}
+    \hat{y} = \sum_{m=1}^{M} \alpha_m \hat{y}_m,
+\end{equation}
+where $\hat{y}_m$ is the prediction of the $m$-th weak learner and $\alpha_m$ is the weight assigned to the $m$-th weak learner.
+While there are various choices for weak learners, decision trees are a common choice.
+Decision trees are a core component of gradient boosting methods. 
+They partition the data into subsets based on feature values, aiming to create groups where data points have similar predicted outcomes. 
+The quality of a split is measured using various metrics. Some common metrics include impurity, information gain, and Gini index. 
+
+Once optimal splits are identified, the tree is constructed by repeatedly partitioning the data until a stopping criterion is met.
+The final model consists of splits that create distinct regions, each with a predicted response value based on the mean of the observations in that region.
+
+This leads us to \textit{Gradient Boosting}.
+Instead of building a single decision tree, gradient boosting constructs multiple trees sequentially, with each new tree correcting the errors of the previous ones.
```

Comment:
Might want to elaborate a bit on how it does this? A sentence or two maybe. It's a bit vague right now and I think it's a core concept in gradient boosting.

Reply:
Elaborating here could also explain why it has to be done sequentially.
Obviously, the reason is that the successor depends on the residuals of the predecessor.

It works by fitting the new predictor to the residual errors made by the previous predictor. Here are the basic steps:
```mermaid
flowchart TD
	nm[Start with a naive model]
	mp[Make predictions]
	cl[Calculate loss]
	tnm[Train new model]
	add[Add new model to ensemble]

	nm --> mp
	mp --> cl
	cl --> tnm
	tnm --> add
	add --> mp
```

So basically, we:
1. Start with an initial model – e.g. it predicts a constant value (could be avg. val or `0.5`)
2. Compute gradients – for each instance in the dataset, we compute the gradients of the loss function w.r.t the prediction of the current ensemble (which is initially just the constant)
3. Then we fit a weak learner to the gradients
4. We update the model by adding the new weak learner, and we usually weigh it by a factor, e.g. a learning rate.
5. We repeat steps 2-4 for a fixed number of iterations or until convergence.

Also, you haven't really explained boosting, which seems like a critical piece? Since you already spent time addressing ensembles & decision trees. It would make it easier to explain GBR.

Boosting was originally called hypothesis boosting.
It refers to any ensemble method that combines several weak learners into a strong learner.

The idea is that we can build a strong predictive model by combining several weak models, e.g. decision trees. We train the predictors sequentially, each one trying to correct its predecessor.
A weak model isn't a poorly performing one, but rather, it's a model that does slightly better than random guessing.

The way it works is that we:
1. Start with an initial model, which is a weak model we fit to the data.
2. We calculate the errors / residuals of this initial model.
3. Then we fit another weak model that focuses on the errors made by the previous model.
4. We combine these weak models through weighted averaging.
5. And finally, we repeat steps 2-4 until a specific number of models are built, or no further improvements can be made.

So each of the models are different from the previous one in the sense that the 'next' weak learner tries to 'fix' the errors the previous one made.

You can imagine you have $x$ features, and you are trying to learn some prediction function $F(x)$.

Boosting is where you learn $F(x)$ as the sum of $M$ weak learners.

$$
F(x)=\sum^{M}_{i=1}f_{i}(x)
$$

where $f_{i}(x)$ is the weak learner.

Gradient boosting generalized boosting by allowing optimizations over arbitrary loss functions. It uses gradient descent to find the minimum of the loss function. And since it can optimize any differentiable loss function, it can be adapted to solve a wide variety of tasks.

You can see how GBR works in this simplified example:

```python
import numpy as np
from sklearn.tree import DecisionTreeRegressor

# Generate noisy quadratic dataset
np.random.seed(42)
X = np.random.rand(100, 1) - 0.5
y = 3 * X[:, 0] ** 2 + 0.05 * np.random.randn(100)

# GBRT
tree_reg1 = DecisionTreeRegressor(max_depth=2, random_state=42)
tree_reg1.fit(X, y)

y2 = y - tree_reg1.predict(X)
tree_reg2 = DecisionTreeRegressor(max_depth=2, random_state=43)
tree_reg2.fit(X, y2) # train on residuals made by the first predictor

y3 = y2 - tree_reg2.predict(X)
tree_reg3 = DecisionTreeRegressor(max_depth=2, random_state=44)
tree_reg3.fit(X, y3) # train on residuals made by the second predictor

# 'Test' set
X_new = np.array([[-0.4], [0.], [0.5]])
sum(tree.predict(X_new) for tree in (tree_reg1, tree_reg2, tree_reg3))
> array([0.49484029, 0.04021166, 0.75026781])
```

I also have a ton of notes describing the gradient boosting **trees** aspect, but this comment is already getting long...